### ISSUE TITLE
regular crashes (due to undefined variables)

### ISSUE THREAD

**Issue by wighawag** [ConsenSys-archive/ganache-cli-archive#378]
## Steps to Reproduce (for bugs)

1. Launch testrpc (I used ```testrpc -i 888 -s seed -b 40``` with 7 ```--account```
2. use it via metamask (with regular request like :
```
eth_getBalance
eth_getTransactionCount
eth_getBalance
eth_getTransactionCount
eth_getBalance
eth_getTransactionCount
eth_getBalance
eth_getTransactionCount
eth_getBlockByNumber
eth_getBlockByNumber
eth_newFilter
eth_getFilterLogs
eth_uninstallFilter
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_call
eth_call
eth_call
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_call
eth_getBalance
eth_getTransactionCount
eth_sendRawTransaction
eth_getTransactionReceipt
eth_getBlockByNumber
eth_getTransactionReceipt
eth_getTransactionReceipt
eth_getBlockByNumber
eth_getBlockByNumber
eth_getTransactionReceipt
eth_getBlockByNumber
eth_getTransactionReceipt
eth_getTransactionReceipt
eth_getTransactionReceipt
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getTransactionReceipt
eth_getBlockByNumber
eth_getTransactionReceipt
eth_getTransactionReceipt
eth_getTransactionReceipt
eth_getBlockByNumber
eth_getBlockByNumber
eth_getTransactionReceipt
eth_getBlockByNumber
eth_getTransactionReceipt
eth_getTransactionReceipt
eth_getTransactionReceipt
eth_getBlockByNumber
eth_getBlockByNumber
eth_getTransactionReceipt
eth_getBlockByNumber
eth_getBlockByNumber
eth_getTransactionReceipt
eth_getTransactionReceipt
eth_getTransactionReceipt
eth_getBlockByNumber
eth_getBlockByNumber
eth_getTransactionReceipt
eth_getBlockByNumber
eth_getTransactionReceipt
eth_getTransactionReceipt
eth_getBlockByNumber
eth_getTransactionReceipt
eth_getBlockByNumber
eth_getBlockByNumber
eth_getTransactionReceipt
eth_getBlockByNumber
eth_getTransactionReceipt

  Transaction: 0x84ac965deb38a0f739ed3c12bf2f47c8c9f6ea52bbc524cbedee7df73e32d556
  Gas usage: 0x01305d
  Block Number: 0x18
  Block Time: Thu Aug 31 2017 06:52:17 GMT+0100 (GMT Daylight Time)

eth_getTransactionReceipt
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBalance
eth_getTransactionCount
eth_getBalance
eth_getTransactionCount
eth_getBalance
eth_getTransactionCount
eth_getBalance
eth_getTransactionCount
eth_getBalance
eth_getTransactionCount
eth_getBalance
eth_getTransactionCount
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBalance
eth_getTransactionCount
eth_getBalance
eth_getTransactionCount
eth_getBalance
eth_getTransactionCount
eth_getBalance
eth_getTransactionCount
eth_getBalance
eth_getTransactionCount
eth_getBlockByNumber
eth_getBlockByNumber
eth_newFilter
eth_getFilterLogs
eth_uninstallFilter
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_call
eth_call
eth_call
eth_getBalance
eth_getTransactionCount
eth_sendRawTransaction
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_call
eth_getBalance
eth_getTransactionCount
eth_sendRawTransaction
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getTransactionReceipt
eth_sendRawTransaction
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getTransactionReceipt
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
```

3. crashes with 
```
C:\Users\wighawag\AppData\Roaming\npm\node_modules\ethereumjs-testrpc\node_modules\solc\soljson.js:1
(function (exports, require, module, __filename, __dirname) { var Module;if(!Module)Module=(typeof Module!=="undefined"?Module:null)||{};var moduleOverrides={};for(var key in Module){if(Module.hasOwnProperty(key)){moduleOverrides[key]=Module[key]}}var ENVIRONMENT_IS_WEB=typeof window==="object";var ENVIRONMENT_IS_WORKER=typeof importScripts==="function";var ENVIRONMENT_IS_NODE=typeof process==="object"&&typeof require==="function"&&!ENVIRONMENT_IS_WEB&&!ENVIRONMENT_IS_WORKER;var ENVIRONMENT_IS_SHELL=!ENVIRONMENT_IS_WEB&&!ENVIRONMENT_IS_NODE&&!ENVIRONMENT_IS_WORKER;if(ENVIRONMENT_IS_NODE){if(!Module["print"])Module["print"]=function print(x){process["stdout"].write(x+"\n")};if(!Module["printErr"])Module["printErr"]=function printErr(x){process["stderr"].write(x+"\n")};var nodeFS=require("fs");var nodePath=require("path");Module["read"]=function read(filename,binary){filename=nodePath["normalize"](filename);va

TypeError: Cannot read property 'gasPrice' of undefined
    at priceSort (C:\Users\wighawag\AppData\Roaming\npm\node_modules\ethereumjs-testrpc\lib\blockchain_double.js:255:61)
    at _siftdown (C:\Users\wighawag\AppData\Roaming\npm\node_modules\ethereumjs-testrpc\node_modules\heap\lib\heap.js:240:11)
    at heappush (C:\Users\wighawag\AppData\Roaming\npm\node_modules\ethereumjs-testrpc\node_modules\heap\lib\heap.js:67:12)
    at Heap.push (C:\Users\wighawag\AppData\Roaming\npm\node_modules\ethereumjs-testrpc\node_modules\heap\lib\heap.js:295:14)
    at BlockchainDouble.sortByPriceAndNonce (C:\Users\wighawag\AppData\Roaming\npm\node_modules\ethereumjs-testrpc\lib\blockchain_double.js:281:12)
    at BlockchainDouble.processNextBlock (C:\Users\wighawag\AppData\Roaming\npm\node_modules\ethereumjs-testrpc\lib\blockchain_double.js:295:8)
    at C:\Users\wighawag\AppData\Roaming\npm\node_modules\ethereumjs-testrpc\lib\statemanager.js:427:21
    at Object.async.whilst (C:\Users\wighawag\AppData\Roaming\npm\node_modules\ethereumjs-testrpc\node_modules\async\lib\async.js:792:13)
    at StateManager.processBlocks (C:\Users\wighawag\AppData\Roaming\npm\node_modules\ethereumjs-testrpc\lib\statemanager.js:416:9)
    at Timeout._onTimeout (C:\Users\wighawag\AppData\Roaming\npm\node_modules\ethereumjs-testrpc\lib\statemanager.js:131:10)
```

4.

## Your Environment

* Version used: 3.0.5
* Environment name and version (e.g. PHP 5.4 on nginx 1.9.1): 
* Server type and version:
* Operating System and version: Windows 10
* Link to your project:

**Comment by benjamincburns** [comment]
@wighawag is this still an issue? Sorry it's taken us so long to get back to you!

v3.0.5 was a long time ago, and I haven't seen anything like this apart from #417, so please reopen if this is still a problem!

**Comment by wighawag** [comment]
Hi @benjamincburns thanks for getting back. I ll let you know if I can encounter the issue with newest version.

**Comment by beether** [comment]
I'm getting this error now when trying to mine multiple transactions in one block.

I submit `miner_stop`, submit transactions, then `evm_mine`, then `miner_start`.  The submitted transactions never complete, and upon submitting the next transaction this error occurs.

When edited it so I only do two transactions in one block, the block gets mined but an error occurs immediately after:

TypeError: callback is not a function

**Comment by benjamincburns** [comment]
@beether can you please raise a separate issue for this problem? I believe that is different from the one originally reported here.

**Comment by benjamincburns** [comment]
@beether actually, it looks like this is already reported. See #383.

**Comment by benjamincburns** [comment]
The issue @wighawag originally reported has now been fixed (see #293 ) and this fix will go out with our next release. Apologies for closing this one prematurely. It seems this has been a bug for quite a long time!
